folding-schemes/src/frontend/noname/mod.rs

```diff
@@ -0,0 +1,201 @@
+use crate::Error;
+use ark_noname::sonobe::NonameSonobeCircuit;
```

Comment:
Seems weird to have the logic of `NonameSonobeCircuit` in a different repo (https://github.com/dmpierre/ark-noname/tree/caf7585aac5cc92fc319d649b0b564a8d9899fcf/src/sonobe.rs#L16), I was expecting this logic to be in the current file (`folding-schemes/src/frontend/noname/mod.rs`), but I'm not sure if I'm missing something.
Which would be the reason to have some sonobe specific logic in a separated `ark-noname`?

Reply:
This is similar to what we do with [`CircomCircuit`](https://github.com/privacy-scaling-explorations/sonobe/blob/d2db27d434dd70c5078cbd1738a9e8e6e0577c4d/folding-schemes/src/frontend/circom/mod.rs#L4), a struct from the `ark-circom` crate. However, since I have the hand on this crate, I created a separate module, specifically for sonobe :)

Reply:
I'm unsure: ark-circom was an already existing repo, which we updated to allow to work with other curves than BN254, and to support allocating the inputs, but there is nothing specifically related to Sonobe. And ideally we will end up opening a PR to the original ark-circom repo integrating those changes (which don't include any Sonobe specific stuff).
Whereas in ark-noname seems that there are specific Sonobe structs, which is what was making me think that we're splitting the Sonobe-related logic across the two different repos (also making ark-noname repo not Sonobe agnostic). Is that this way? (I've might be missing stuff)

Reply:
The way we allocate the inputs is the same both in `ark-circom` and in `ark-noname`. In `ark-circom` the struct to do this is `CircomCircuit`, in `ark-noname` the struct is `NonameSonobeCircuit`. 
I would argue that, in both cases, the way we allocate inputs is specific to sonobe: another folding lib could have a different strategy for how it chooses to allocate those. That is why [what was merged in the noname repo](https://github.com/zksecurity/noname/pull/137) only consists in the arkworks-to-noname logic: it does not include the "specific" way we allocate inputs when working with sonobe. 
I guess that will be the same if the work done on the `ark-circom` repo is ever merged back upstream: it will most likely only include the work done on making it curve agnostic, but not the work we did to make it work with sonobe.
If that is confusing, I can change the name of the struct in the `ark-noname` lib to something that does not mention sonobe at all?